### PR TITLE
Store node instead of index for starting state

### PR DESCRIPTION
**Pull request by definelicht**
There was a bug where `sdfg._start_state` was stored as an index, but read as a node. I've changed it to be stored as a node, since this is more robust to the SDFG changing.

### CODE REVIEW

File: dace/sdfg.py [hunk 1]
```diff
@@ -413,17 +413,21 @@ def start_state(self):
             return source_nodes[0]
         if self._start_state is None:
             raise ValueError('Ambiguous or undefined starting state for SDFG')
+        if self._start_state not in source_nodes:
+            raise ValueError("Start state {} not found in source nodes".format(
+                self._start_state))
+        return self._start_state
```
Comment (tbennun) [inline dace/sdfg.py]: What if the graph is one state in a loop and an exit? The start state will not be a source node..

Comment (definelicht) [reply dace/sdfg.py]: The only way around that as far as I can tell is to allow _any_ state to be a start state. I doubt our codegen supports that?

Comment (tbennun) [reply dace/sdfg.py]: I think this is what _start_state is for - to allow a non-source-node to be a starting state

Comment (definelicht) [reply dace/sdfg.py]: Not necessarily, it's possible to have multiple source nodes, which is the situation where you'd need user input to decide where to start (not that this ever happens in practice, as far as I know).

**Pull request merged**